src/braket/default_simulator/linalg_utils.py

```diff
@@ -17,7 +17,48 @@
 import numpy as np
 
 
-def multiply_matrix(state: np.ndarray, matrix: np.ndarray, targets: Tuple[int, ...]) -> np.ndarray:
+def apply_operation(
+    state: np.ndarray,
+    matrix: np.ndarray,
+    targets: Tuple[int, ...],
+    controls: Optional[Tuple[int]] = (),
+    control_state: Optional[Tuple[int]] = (),
+) -> np.ndarray:
+    """Applies the given matrix to the given state, controlling the operation as specified.
+
+    Args:
+        state (np.ndarray): The state to multiply the matrix by.
+        matrix (np.ndarray): The matrix to apply to the state.
+        targets (Tuple[int]): The qubits to apply the state on.
+        controls (Optional[Tuple[int]]): The qubits to control the operation on. Default ().
+        control_state (Optional[Tuple[int]]): A tuple of same length as `controls` with either
+            a 0 or 1 in each index, corresponding to whether to control on the |0⟩ or |1⟩ state.
+            Default (1,) * len(controls).
+
+    Returns:
+        np.ndarray: The state after the matrix has been applied.
+    """
+    if not controls:
+        return multiply_matrix(state, matrix, targets)
+
+    control_state = control_state or (1,) * len(controls)
+    num_qubits = len(state.shape)
+    slices = (slice(None, 1), slice(1, None), slice(None, None))
```

Comment:
Still a super-super nit: this can be defined _outside_ of the function so it doesn't need to even be instantiated every time.